galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+
+    def path_difference(self, nx, scale, theta_x=0.0*galsim.degrees, theta_y=0.0*galsim.degrees):
+        """ Compute effective pathlength differences due to phase screen.
+
+        @param nx       Size of output array
+        @param scale    Scale of output array pixels in meters
+        @param theta_x  Field angle corresponding to center of output array.
+        @param theta_y  Ditto.
+        @returns   Array of pathlength differences in nanometers.  Multiply by 2pi/wavelength to get
+                   array of phase differences.
+        """
+        img = galsim.Image(np.ascontiguousarray(self.screen), scale=self.screen_scale)
+        ii = galsim.InterpolatedImage(img, calculate_stepk=False, calculate_maxk=False,
+                                      normalization='sb')
+        ii = ii.shift(1000*self.altitude * theta_x.tan(), 1000*self.altitude * theta_y.tan())
+        return ii.drawImage(nx=nx, ny=nx, scale=scale, method='sb', dtype=np.float64).array
+
+    def reset(self):
+        """Reset phase screen back to time=0."""
+        self.rng = copy.deepcopy(self.orig_rng)
+        fx, fy = self._freq()
+        self._init_screen(fx, fy)
+
+
+class PhaseScreenList(object):
+    """ List of phase screens that can be turned into a PSF.  Screens can be either atmospheric
+    layers or optical phase screens.  Generally, one would assemble a PhaseScreenList object using
+    the function `Atmosphere`.  Layers can be added, removed, appended, etc. just like items can be
+    manipulated in a python list.  For example:
+
+    # Create an atmosphere with three layers, two of which are frozen and one not frozen.
+    > screens = galsim.PhaseScreenList([galsim.FrozenAtmosphericScreen(...),
+                                        galsim.FrozenAtmosphericScreen(...),
+                                        galsim.ARAtmosphericScreen(...)])
+    # Add another layer
+    > screens.append(galsim.FrozenAtmosphericScreen(...))
+    # Remove the second layer
+    > del screens[1]
+    # Switch the first and second layer.  Silly, but works...
+    > screens[0], screens[1] = screens[1], screens[0]
+
+    Note that creating new PhaseScreenLists from old PhaseScreenLists copies the wrapped phase
+    screens by reference, not value.  Thus, advancing the screens in one list will also advance the
+    screens in a copy:
+
+    > more_screens = screens[0:2]
+    > more_screens.advance()
+    > assert more_screens[0] == screens[0]
+
+    > more_screens = galsim.PhaseScreenList(screens)
+    > screens.reset()
+    > psf = screens.getPSF(exptime=exptime, ...)        # starts at t=0
+    > psf2 = more_screens.getPSF(exptime=exptime, ...)  # starts at t=exptime
+    > assert psf != psf2
+
+    Methods
+    -------
+    getPSF()           Obtain a PSF from this set of phase screens.  See PhaseScreenPSF docstring
+                       for more details.
+    getPSFs()          Optimized method for obtaining multiple PSFs from a single PhaseScreenList.
```

Comment:
Looks like this functionality is now part of `getPSF`.  Also, recommend changing the verb here to make or build or generate, rather than get.  Usually get refers to something that is already stored in the object so is trivial to return.  This requires a bit of computation to construct the PSF.  And moreover, the return value here is not the same for repeated calls, which would normally be the case for a getter.